src/coord/src/coord.rs

```diff
@@ -867,6 +903,93 @@ impl<S: Append + 'static> Coordinator<S> {
             .unwrap();
     }
 
+    /// Attempts to commit all pending write transactions in a group commit. If the timestamp
+    /// chosen for the writes is not ahead of `now`, then we can execute and commit the writes
+    /// immediately. Otherwise we must wait for `now` to advance past the timestamp chosen for the
+    /// writes.
+    async fn try_group_commit(&mut self, timestamp: Option<Timestamp>) {
+        if self.pending_writes.is_empty() {
+            return;
+        }
+
+        let timestamp = if let Some(timestamp) = timestamp {
+            timestamp
+        } else {
+            self.peek_local_ts()
+        };
+        let now = (self.catalog.config().now)();
+
+        if timestamp > now {
+            // Cap retry time to 1s. In cases where the system clock has retreated by
+            // some large amount of time, this prevents against then waiting for that
+            // large amount of time in case the system clock then advances back to near
+            // what it was.
+            let remaining_ms =
+                std::cmp::min(timestamp.saturating_sub(now).saturating_add(1), 1_000);
```

Comment:
@mjibson I'm not entirely clear what `saturating_add` is for here if you could provide some insight.

Reply:
Just a way to specify what to do in case of overflow. I think default is to wrap, which is never what we want here.

Reply:
(The question was about the 1, not the saturating add, and it was because 11105 needed to wait for the next timestamp, which this one doesn't.)

Reply:
Make sense, I removed the addition.